rasa/engine/validation.py

```diff
@@ -0,0 +1,342 @@
+import inspect
+import logging
+import typing
+from typing import Optional, Callable, Text, Tuple, Dict, Type, Any, Set, Union, TypeVar
+
+import dataclasses
+
+import rasa.utils.common
+import typing_utils
+
+from rasa.engine.exceptions import GraphSchemaValidationException
+from rasa.engine.graph import (
+    GraphSchema,
+    GraphComponent,
+    SchemaNode,
+    Fingerprintable,
+    ExecutionContext,
+)
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+
+
+@dataclasses.dataclass
+class ParameterInfo:
+    """Stores metadata about a function parameter."""
+
+    type_annotation: Union[TypeVar, Text, Type]
```

Comment:
Might be worth having `Union[TypeVar, Text, Type]` as a variable now we use it quite often.

Reply:
done